pkg/controller/storagecluster/reconcile.go

```diff
@@ -654,29 +703,37 @@ func (r *ReconcileStorageCluster) ensureCephCluster(sc *ocsv1.StorageCluster, re
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: cephCluster.Name, Namespace: cephCluster.Namespace}, found)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating CephCluster")
+			reqLogger.Info(fmt.Sprintf("Creating %s", cephClusterTypeStr))
 			return r.client.Create(context.TODO(), cephCluster)
 		}
 		return err
 	}
 
 	// Update the CephCluster if it is not in the desired state
 	if !reflect.DeepEqual(cephCluster.Spec, found.Spec) {
-		reqLogger.Info("Updating spec for CephCluster")
-		// Check if Cluster is Expanding
-		if len(found.Spec.Storage.StorageClassDeviceSets) < len(cephCluster.Spec.Storage.StorageClassDeviceSets) {
-			r.phase = statusutil.PhaseClusterExpanding
-		} else if len(found.Spec.Storage.StorageClassDeviceSets) == len(cephCluster.Spec.Storage.StorageClassDeviceSets) {
-			for _, countInFoundSpec := range found.Spec.Storage.StorageClassDeviceSets {
-				for _, countInCephClusterSpec := range cephCluster.Spec.Storage.StorageClassDeviceSets {
-					if countInFoundSpec.Name == countInCephClusterSpec.Name && countInCephClusterSpec.Count > countInFoundSpec.Count {
-						r.phase = statusutil.PhaseClusterExpanding
+		reqLogger.Info(fmt.Sprintf("Updating spec for %s", cephClusterTypeStr))
+		if sc.Spec.ExternalStorage.Enable {
+			sc.Status.Phase = string(found.Status.State)
```

Comment:
Can you explain what this does? I think I have an idea, but I'm not quite sure. A comment in the code would be helpful.

Reply:
Still missing a comment here.

Reply:
In an external cluster mode, (and when ideal cephcluster.Spec doesn't match with the found cephCluster.Spec) we don't have much to do. Update the `Status.Phase` with the `State` of the found cluster.

@umangachapagain , will be able to explain better...

Reply:
Added the comments...

Reply:
You didn't mention why, though. Why are you copying State into Phase?

Reply:
This is very confusing at first look. I think what you want to do here is,
1. If it is an Update request for External Ceph Cluster just proceed with the update
2. If it is an Update request for Ceph Cluster,
       2.1 Check if Cluster is expanding and set PhaseExpanding accordingly. Then continue normal update
       2.2 If it is not expanding, proceed with normal update to reach desired state